src/task/MonitorTask.ts

```diff
@@ -438,4 +462,42 @@ export default class MonitorTask
       poorConnectionCount: this.context.poorConnectionCount,
     };
   };
+
+  /**
+   * Degrade video codec to an alternative codec
+   */
+  private degradeVideoCodec(cause: string): void {
+    // Degrade video codec if there are other codec options and current codec is not H264 CBP or VP8
+    if (
+      this.context.meetingSupportedVideoSendCodecPreferences !== undefined &&
+      this.context.meetingSupportedVideoSendCodecPreferences.length > 1 &&
+      !(
+        this.context.meetingSupportedVideoSendCodecPreferences[0].equals(
+          VideoCodecCapability.h264ConstrainedBaselineProfile()
+        ) ||
+        this.context.meetingSupportedVideoSendCodecPreferences[0].equals(VideoCodecCapability.vp8())
+      )
+    ) {
+      const newMeetingSupportedVideoSendCodecPreferences: VideoCodecCapability[] = [];
+      for (const capability of this.context.videoSendCodecPreferences) {
+        if (!capability.equals(this.context.meetingSupportedVideoSendCodecPreferences[0])) {
+          newMeetingSupportedVideoSendCodecPreferences.push(capability);
+        }
+      }
+      if (newMeetingSupportedVideoSendCodecPreferences.length > 0) {
+        this.context.logger.info(
+          `Downgrading codec from ${this.context.meetingSupportedVideoSendCodecPreferences[0].codecName} to ${newMeetingSupportedVideoSendCodecPreferences[0].codecName} due to ${cause}`
+        );
+        this.context.videoSendCodecsBlocklisted.push(
+          this.context.meetingSupportedVideoSendCodecPreferences[0]
+        );
+        this.context.meetingSupportedVideoSendCodecPreferences = newMeetingSupportedVideoSendCodecPreferences;
+        this.context.audioVideoController.update({ needsRenegotiation: true });
+      } else {
+        this.context.logger.warn(
+          'Degrading video codec failed since there is no alternative codec to select'
```

Comment:
Nit: This may flood the client logs (every second) if the policy evaluates to minimum health consecutively due to some issue. do we need this at warn level?

Reply:
Actually, it's very unlikely this log will be triggered since the condition is fairly strict: 
1. Current `meetingSupportedVideoSendCodecPreferences` should at least have two codecs (normally, the updated `meetingSupportedVideoSendCodecPreferences` will just include current codecs except the first one);
2. Current codec is not H.264 CBP or VP8;
3. Excluding current codec, there should be no overlap between `meetingSupportedVideoSendCodecPreferences` and `videoSendCodecPreferences` (note that `meetingSupportedVideoSendCodecPreferences` is basically derived from `videoSendCodecPreferences`, this is even less likely to happen).

Given the the overlay of these conditions should be extremely rare (if even possible), I don't think this will really flood client log unless something goes very wrong.